app/src/main/java/com/example/albumsapp/adapter/PhotosAdapter.kt

```diff
@@ -1,37 +1,46 @@
 package com.example.albumsapp.adapter
 
+
+import android.graphics.Color.parseColor
 import android.view.LayoutInflater
 import android.view.View
 import android.view.ViewGroup
-import android.widget.Button
-import androidx.navigation.findNavController
+import android.widget.ImageView
 import androidx.recyclerview.widget.RecyclerView
-import com.example.albumsapp.PhotosListFragmentDirections
 import com.example.albumsapp.R
 import com.example.albumsapp.model.Photo
-import com.example.albumsapp.viewmodel.PhotoViewModel
+import com.squareup.picasso.Picasso
+import java.util.*
+
 
-class PhotosAdapter : RecyclerView.Adapter<PhotosAdapter.PhotosViewHolder>() {
+class PhotosAdapter(
+    private val listOfPhotos: List<Photo>,
+    private val onPhotoClicked: (Photo) -> Unit
+) : RecyclerView.Adapter<PhotosAdapter.PhotosViewHolder>() {
 
-    private val viewModel: PhotoViewModel = PhotoViewModel()
-    private val listOfPhoto: List<Photo>? = viewModel.photos.value
     class PhotosViewHolder(view: View) : RecyclerView.ViewHolder(view) {
-        val button = view.findViewById<Button>(R.id.button_item)
+        val imageView: ImageView = view.findViewById(R.id.album_photo)
     }
 
     override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): PhotosViewHolder {
         val layout = LayoutInflater.from(parent.context)
-            .inflate(R.layout.item_view, parent, false)
+            .inflate(R.layout.grid_view_photo, parent, false)
 
         return PhotosViewHolder(layout)
     }
 
     override fun onBindViewHolder(holder: PhotosViewHolder, position: Int) {
-        holder.button.setOnClickListener {
-            val action = PhotosListFragmentDirections.actionListOfPhotosToSelectedPhoto()
-            holder.itemView.findNavController().navigate(action)
+        val item = listOfPhotos[position]
+        var color = "#" + item.thumbnailUrl.takeLastWhile { it != '/' }.uppercase(Locale.ROOT)
+        if (color.length < 7) {
+            color += "0"
+        }
```

Comment:
Also, let's create a separate class for example `PhotoColorParser` and move color parsing there. Otherwise we brake the Single Responsibility principle. Adapter should only _adapt_ items to view, and not do anything else (parse colors in our case).